### PR TITLE
style(ui): Reformat and standardize all RadUrns

### LINKED ISSUE #643: Reformat and standardize all RadUrns
## To do
Make the design match [this component](https://www.figma.com/file/owmgsbs6lnUt8R1bixstCA/%F0%9F%8F%97-Components?node-id=3372%3A10405).

Screenshot 2020-07-09 at 15 15 04

- [x] Standardize the component across the app so it always looks the same
- [x] the RadID should show the first 6 and last 6 characters and truncate from in the middle
- [x] always has a background
- [x] always has rounded corners
- [x] same font size and colors
- [x] always copyable on hover

---

## Places where RadID is displayed
- [x] Clone overlay
image
- [x] Project ID
image
- [x] Profile ID
image
- [x] Discover page project tiles
image
- [x] Org ID
image
- [x] Profile overflow ••• menu
image
- [x] Project overflow ••• menu
image
- [x] Transaction id
image
- [x] Wallet send screen
image

### PR DESCRIPTION
**Pull request by juliendonck**
closes #643 

- has `showOnHover` bool to show the full urn on hover if added to the component (made it conditional cause it often expands of the screen)
- removes `rad:git:` when displayed but it's still there when you copy it.

### CODE REVIEW

**Comment by juliendonck** [comment]
@brandonhaslegs moving the copy icon will be weird I think. It will look like it represents the string after it. I think it moving would be a bigger issue if you had to click on that to copy, but since you can click anywhere to copy, it doesn't matter as much.

File: ui/DesignSystem/Component/Urn.svelte [hunk 1]
```diff
@@ -1,25 +1,91 @@
...
+  export let showOnHover = false;
+
+  let cleanUrn;
+  if (urn.includes("%rad:git:")) {
+    cleanUrn = urn.substring(9, urn.length);
+  } else if (urn.includes("rad:git:")) {
+    cleanUrn = urn.substring(8, urn.length);
+  } else {
+    cleanUrn = urn;
+  }
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Urn.svelte:15]:
```suggestion
const cleanUrn = urn.replace(/^%?rad:git:/, "");
```

```javascript
"%rad:git:hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac".replace(/^%?rad:git:/, "")
  → "hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac"

"rad:git:hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac".replace(/^%?rad:git:/, "")
  → "hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac"

"hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac".replace(/^%?rad:git:/, "")
  → "hwd1yrenfedtqo5cep373tkpf3f4474rx5jqeqazhpmfyusujnsftcpbpac"
```

File: ui/DesignSystem/Component/Urn.svelte [hunk 2]
```diff
@@ -1,25 +1,91 @@
...
+
+  const firstSix = cleanUrn.substring(0, 7);
+  const lastSix = cleanUrn.substring(cleanUrn.length - 7, cleanUrn.length);
+
+  const fullUrn = { className: "hidden" };
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Urn.svelte:20]: Why does this need to be an object?

File: ui/DesignSystem/Component/Wallet/Receive.svelte [hunk 3]
```diff
@@ -24,5 +28,7 @@
     </Text>
   {/if}
   <Title style="padding: 1.5rem 0 0.5rem 0;">Address</Title>
-  <Urn urn={accountId} />
+  <div class="center">
```
Comment (rudolfs) [inline ui/DesignSystem/Component/Wallet/Receive.svelte:31]: We have `<Flex>` for that.

File: ui/Screen/DesignSystemGuide.svelte [hunk 4]
```diff
@@ -1074,6 +1075,10 @@
       <Swatch>
         <Stats branches={2} commits={12} contributors={4} />
       </Swatch>
+
+      <Swatch>
+        <Urn urn="5FA9nQDVg267DEd8m1ZypXLBnvN7SFxYwV7ndqSYGiN9TTpu" />
+      </Swatch>
```
Comment (rudolfs) [inline ui/Screen/DesignSystemGuide.svelte:1081]: Could also add the other variants here.

File: ui/DesignSystem/Component/Urn.svelte (hunk 1 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Urn.svelte]: Nice one https://github.com/radicle-dev/radicle-upstream/pull/731/commits/3d8296ee23532210fff2e062b3b89920f21d013e

File: ui/DesignSystem/Component/Urn.svelte (hunk 2 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Urn.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/731/commits/e4bf58400de29efcb6703c0cb0eb17f83ee78f7f

File: ui/DesignSystem/Component/Wallet/Receive.svelte (hunk 3 above)
Comment (juliendonck) [reply ui/DesignSystem/Component/Wallet/Receive.svelte]: https://github.com/radicle-dev/radicle-upstream/pull/731/commits/26a4ef75ad98cfb544650ab690ce9aa8ee106a80

File: ui/Screen/DesignSystemGuide.svelte (hunk 4 above)
Comment (juliendonck) [reply ui/Screen/DesignSystemGuide.svelte:1081]: https://github.com/radicle-dev/radicle-upstream/pull/731/commits/4a9b3c5116431a32169b623ededfdef1be279cc0

**Comment by rudolfs** [comment]
Another small suggestion:

```diff
diff --git a/ui/DesignSystem/Component/Copyable.svelte b/ui/DesignSystem/Component/Copyable.svelte
index 16446b9..c606432 100644
--- a/ui/DesignSystem/Component/Copyable.svelte
+++ b/ui/DesignSystem/Component/Copyable.svelte
@@ -12,14 +12,23 @@
   let slotContent;
   let copyIcon = iconBeforeCopy;

+  let copied = false;
+
   const copy = () => {
+    if (copied) {
+      return
+    }
+
     const content =
       copyContent !== null ? copyContent : slotContent.textContent;
     copyToClipboard(content.trim());

+    copied = true;
+
     copyIcon = Icon.Check;
     setTimeout(() => {
       copyIcon = Icon.Copy;
+      copied = false;
     }, 1000);
   };
 </script>
```

Before:
no-debounce

After:
debounce

**Comment by rudolfs** [comment]
The copy icon state is not shared between expanded and non-expanded urn:
bug

**Comment by juliendonck** [comment]
@rudolfs addressed both your comment. Made it simpler as well, great feedback :)

**Comment by rudolfs** [comment]
@juliendonck now you just have to remove all the unused imports that eslint is complaining about on CI.

**Pull request merged**